internal/cmd/launcher.go

```diff
@@ -283,31 +282,32 @@ func analyze(gs *state.GlobalState, args []string) (k6deps.Dependencies, error)
 		Manifest:  k6deps.Source{Ignore: true},
 	}
 
-	if len(args) == 0 {
-		gs.Logger.
-			Debug("The command did not receive an input script.")
-		return nil, errScriptNotFound
+	if len(args) < 1 {
+		return nil, fmt.Errorf("the invoked command needs a file path or pass the test via Stdin")
 	}
 
-	scriptname := args[0]
-	if scriptname == "-" {
-		gs.Logger.
-			Debug("Test script provided by Stdin is not yet supported from Binary provisioning feature.")
-		return nil, errUnsupportedFeature
+	sourceRootPath := args[0]
+	gs.Logger.WithField("source", "sourceRootPath").
+		Debug("Launcher is resolving and reading the test")
+	src, _, pwd, err := readSource(gs, sourceRootPath)
+	if err != nil {
+		return nil, fmt.Errorf("reading source for analysis %w", err)
 	}
 
-	if _, err := gs.FS.Stat(scriptname); err != nil {
-		gs.Logger.
-			WithField("path", scriptname).
-			WithError(err).
-			Debug("The requested test script's file is not available on the file system.")
-		return nil, errScriptNotFound
+	// if sourceRooPath is stdin ('-') we need to preserve the content
+	if sourceRootPath == "-" {
+		gs.Stdin = bytes.NewBuffer(src.Data)
 	}
 
-	if strings.HasSuffix(scriptname, ".tar") {
-		dopts.Archive.Name = scriptname
+	if strings.HasSuffix(sourceRootPath, ".tar") {
```

Comment:
To facilitate maintenance, I suggest creating a constant for the tar file extension. This could be done in a separate PR though 🙇🏻

Reply:
Tracked here https://github.com/grafana/k6/issues/4987